src/structure.js

```diff
@@ -15,22 +16,22 @@ export default function analyze(Type, value) {
   return flatMap(analyzeType(value), pure(Tree, new Node(Type, [])));
 }
 
-function analyzeType(value) {
+export function analyzeType(value) {
   return (node) => {
-    let InitialType = node.Type;
+    let SugarFreeType = isSugar(node.Type) ? desugar(node.Type) : node.Type;
```

Comment:
Let's keep knowledge of sugaring inside the sugar module. Any reason why:

```js
let InitialType = desugar(node.Type);
```
won't work? if it's already desugard, then `desugar` can just passthrough.

Reply:
yep, good idea